File: sorting/sort.java
import java.util.Random;
import java.util.ArrayList;
import java.util.List;

public class Sort {
    public static List<Integer> quicksort(List<Integer> arr) {
        if (arr.size() <= 1) {
            return arr;
        }
        
        int pivot = arr.get(arr.size() / 2);
        List<Integer> left = new ArrayList<>();
        List<Integer> middle = new ArrayList<>();
        List<Integer> right = new ArrayList<>();
        
        for (int x : arr) {
            if (x < pivot) {
                left.add(x);
            } else if (x > pivot) {
                right.add(x);
            } else {
                middle.add(x);
            }
        }
        
        left = quicksort(left);
        right = quicksort(right);
        
        List<Integer> result = new ArrayList<>();
        result.addAll(left);
        result.addAll(middle);
        result.addAll(right);
        
        return result;
    }
    
    public static void main(String[] args) {
        int size = 1000000;
        Random rand = new Random(42);
        List<Integer> arr = new ArrayList<>();
        
        for (int i = 0; i < size; i++) {
            arr.add(rand.nextInt(1000000) + 1);
        }
        
        List<Integer> sorted = quicksort(arr);
        
        System.out.print("First 10: ");
        for (int i = 0; i < 10; i++) {
            System.out.print(sorted.get(i) + " ");
        }
        System.out.println();
        
        System.out.print("Last 10: ");
        for (int i = size - 10; i < size; i++) {
            System.out.print(sorted.get(i) + " ");
        }
        System.out.println();
    }
}

File: sorting/sort.kt
import java.util.Random

fun quicksort(arr: List<Int>): List<Int> {
    if (arr.size <= 1) {
        return arr
    }
    
    val pivot = arr[arr.size / 2]
    val left = arr.filter { it < pivot }
    val middle = arr.filter { it == pivot }
    val right = arr.filter { it > pivot }
    
    return quicksort(left) + middle + quicksort(right)
}

fun main() {
    val size = 1000000
    val arr = mutableListOf<Int>()
    
    val rand = Random(42)
    repeat(size) {
        arr.add(rand.nextInt(1000000) + 1)
    }
    
    val sorted = quicksort(arr)
    
    print("First 10: ")
    for (i in 0 until 10) {
        print("${sorted[i]} ")
    }
    println()
    
    print("Last 10: ")
    for (i in size - 10 until size) {
        print("${sorted[i]} ")
    }
    println()
}
